ground/src/main/java/com/google/android/ground/repository/LocationOfInterestRepository.kt

```diff
@@ -127,4 +124,33 @@ constructor(
       MutationEntitySyncStatus.IN_PROGRESS,
       MutationEntitySyncStatus.FAILED
     )
+
+  /** Returns a flowable of all [LocationOfInterest] for the currently active [Survey]. */
+  fun getAllLocationsOfInterestOnceAndStream(): Flowable<Set<LocationOfInterest>> =
+    surveyRepository.activeSurvey
+      .switchMap { survey ->
+        survey
+          .map { localLoiStore.getLocationsOfInterestOnceAndStream(it) }
+          .orElse(Flowable.just(setOf()))
+      }
+      .distinctUntilChanged()
+
+  /** Returns a flowable of all [LocationOfInterest] within the map bounds (viewport). */
+  fun getWithinBoundsOnceAndStream(
```

Comment:
Camera bounds sounds like something that belongs in the view or in MapStateRepository. Can we pass that in as an arg here instead of maintaining it in this class?

Reply:
`LatLngBounds` flowable is already passed via an arg. Are you referring to something else?

Reply:
Sorry, I missed that the bounds were being passed in as an arg. Please disregard.